hazelcast/src/main/java/com/hazelcast/spi/impl/merge/AbstractSplitBrainHandlerService.java

```diff
@@ -98,20 +98,14 @@ public void run() {
                 Iterator<Store> iterator = storeIterator(partitionId);
                 while (iterator.hasNext()) {
                     Store store = iterator.next();
-
-                    if (isLocalPartition(partitionId)
-                            && hasEntry(store)
-                            && !hasDiscardPolicy(store)) {
```

Comment:
We called `hasDiscardPolicy(store)`, but it should be given the merge policy. So this check never worked and the `DiscardMergePolicy` triggered merge operations like any other merge policy.

Reply:
congrats, good finding.

is it possible to have a test for this case?

Reply:
We would need to verify, that the `DiscardMergePolicy` has not been called or no merge operations have been sent. I don't think there is an easy way to do this.

I just saw this because I broke the merging of ICache while working on a PR and it also crashed the `DiscardMergePolicy` test, which is normally always green, even if other merge policies are broken.